src/gui/qgsidentifymenu.cpp

```diff
@@ -26,6 +26,7 @@
 #include "qgssettings.h"
 #include "qgsgui.h"
 
+//TODO 4.0 add explicitely qobject parent to constructor
```

Comment:
Does this need to wait for 4.0? I think that can be implemented without API break (see my previous comment)

Reply:
@nyalldawson suggested to wait to 4.0. Wouldn't it be technically API break even if you add parent = nullptr in constructor, because you need to care about destruction?

Reply:
In case you specify `nullptr` for parent, the old behavior will jump in and canvas is used as parent.

Reply:
while this will work, it would be in my opinion very counter-intuitive.

Reply:
ok, let's move ahead then :+1: